pkg/spire/spire.go

```diff
@@ -0,0 +1,75 @@
+/*
+Copyright 2022 The Tekton Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// The spire package is used to interact with the Spire server and Spire agent respectively.
+// The pipeline controller (once registered) is able to create and delete entries in the Spire server
+// for the various TaskRuns that it instantiates. The TaskRun is able to attest to the Spire agent
+// and obtains the valid SVID (SPIFFE Verifiable Identity Document) to sign the TaskRun results.
+// Separately, the pipeline controller SVID is used to sign the TaskRun Status to validate no modification
+// during the TaskRun execution. Each TaskRun result and status is verified and validated once the
+// TaskRun execution is completed. Tekton Chains will also validate the results and status before
+// signing and creating attestation for the TaskRun.
+package spire
+
+import (
+	"context"
+	"time"
+
+	"github.com/tektoncd/pipeline/pkg/apis/pipeline/v1beta1"
+	spireconfig "github.com/tektoncd/pipeline/pkg/spire/config"
+	"go.uber.org/zap"
+	corev1 "k8s.io/api/core/v1"
+)
+
+const (
+	// TaskRunStatusHashAnnotation TaskRun status annotation Hash Key
+	TaskRunStatusHashAnnotation = "tekton.dev/status-hash"
+	// taskRunStatusHashSigAnnotation TaskRun status annotation hash signature Key
+	taskRunStatusHashSigAnnotation = "tekton.dev/status-hash-sig"
+	// controllerSvidAnnotation TaskRun status annotation controller SVID Key
+	controllerSvidAnnotation = "tekton.dev/controller-svid"
+	// NotVerifiedAnnotation TaskRun status annotation not verified by spire key that get set when status match fails
+	NotVerifiedAnnotation = "tekton.dev/not-verified"
```

Comment:
```suggestion
	VerifiedAnnotation = "tekton.dev/spire-verified"
```

A general rule of thumb I like to follow is avoiding double negatives. It helps avoid ambiguous cases like does the absence of this annotation mean that it was successfully verified?

Also recommend using `spire-verified` so this doesn't end up conflicting with other verification checks.

Reply:
Agreed. Changed to suggested.